evap/results/tests/test_exporters.py

```diff
@@ -590,3 +590,68 @@ def test_text_answer_export(self):
         self.assertEqual(sheet.row_values(3)[0], questions[0].text)
         self.assertEqual(sheet.row_values(5)[0], questions[1].text)
         self.assertEqual(sheet.row_values(6)[0], questions[2].text)
+
+    def test_total_average(self):
+        program = baker.make(Program)
+
+        questionnaire_1 = baker.make(Questionnaire, order=1, type=Questionnaire.Type.TOP)
+        questionnaire_2 = baker.make(Questionnaire, order=4, type=Questionnaire.Type.TOP)
+
+        question_1 = baker.make(Question, type=QuestionType.GRADE, questionnaire=questionnaire_1)
+        question_2 = baker.make(Question, type=QuestionType.POSITIVE_LIKERT, questionnaire=questionnaire_2)
+
+        evaluation_1 = baker.make(
+            Evaluation,
+            course__programs=[program],
+            state=Evaluation.State.PUBLISHED,
+            _participant_count=2,
+            _voter_count=2,
+        )
+
+        evaluation_2 = baker.make(
+            Evaluation,
+            course__programs=[program],
+            state=Evaluation.State.PUBLISHED,
+            _participant_count=3,
+            _voter_count=3,
+        )
+
+        evaluation_1.general_contribution.questionnaires.set([questionnaire_1])
+
+        make_rating_answer_counters(question_1, evaluation_1.general_contribution, [1, 1, 0, 0, 0])
+
+        evaluation_2.general_contribution.questionnaires.set([questionnaire_1, questionnaire_2])
+
+        make_rating_answer_counters(question_1, evaluation_2.general_contribution, [1, 2, 0, 0, 0])
+        make_rating_answer_counters(question_2, evaluation_2.general_contribution)
+
+        cache_results(evaluation_1)
+        cache_results(evaluation_2)
+
+        binary_content = BytesIO()
+        ResultsExporter().export(
+            binary_content,
+            [evaluation_1.course.semester, evaluation_2.course.semester],
+            [
+                (
+                    [course_program.id for course_program in evaluation_1.course.programs.all()]
+                    + [course_program.id for course_program in evaluation_2.course.programs.all()],
+                    [evaluation_1.course.type.id, evaluation_2.course.type.id],
+                )
+            ],
+            True,
+            True,
+        )
+        binary_content.seek(0)
+        workbook = xlrd.open_workbook(file_contents=binary_content.read())
+
+        self.assertAlmostEqual(workbook.sheets()[0].row_values(5)[1], 1.5833333333333335)
+
+        self.assertEqual(
+            workbook.sheets()[0].row_values(0)[1],
+            "Average for this question over all evaluations in all published semesters",
+        )
+
+        # test ragged questionaire
+        self.assertEqual(float(workbook.sheets()[0].row_values(8)[1]), float(workbook.sheets()[0].row_values(8)[3]))
+        self.assertEqual("", workbook.sheets()[0].row_values(8)[2])
```

Comment:
```suggestion
        # average for second questionnaire must be the value from evaluation2 (since evaluation1 doesn't have the questionnaire)
        self.assertEqual(float(workbook.sheets()[0].row_values(8)[1]), float(workbook.sheets()[0].row_values(8)[3]))
        # average field next to the questionnaire title must be empty
        self.assertEqual("", workbook.sheets()[0].row_values(8)[2])
```

Reply:
isn't `row_values(8)[2]` the second evaluation (which doesn't use this questionnaire, so the cell must be empty)?